CI: Fix log export configuration

maxknv opened the pull request:
fix creating tables for log export if format has been changed (no cluster creds were exported to do the queries) 

<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)


### Changelog entry (a [user-readable short description](https://github.com/ClickHouse/ClickHouse/blob/master/docs/changelog_entry_guidelines.md) of the changes that goes into CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->

maxknv committed:
Fix credentials export

azat commented:
Hm, I don't see any entries for `trace_log`, though there are entries for `text_log` and others

```
clickhouse-cloud$  :) select count() from merge(default, 'trace_log_') where event_date >= today()-3 and pull_request_number = 87240

SELECT count()
FROM merge(default, 'trace_log_')
WHERE (event_date >= (today() - 3)) AND (pull_request_number = 87240)

Query id: f91c6ec3-ea31-427a-9479-1b783c3b3b09

   ┌─count()─┐
1. │       0 │
   └─────────┘
```

azat commented:
This is because `trace_log` requires table creation, and it does not work

azat committed:
ci/setup_log_cluster: ensure that vars are set

azat committed:
ci/setup_log_cluster: better bash

azat committed:
ci/setup_log_cluster: fix bash magic

- function foo { ... } or foo() { ... } → normal function, runs in the current shell.
- foo() ( ... ) → runs the body in a subshell (like running in (...)).

azat commented on ci/jobs/scripts/functional_tests/setup_log_cluster.sh:
```diff
@@ -14,28 +14,27 @@ set -e
 
 
-function __set_connection_args
+function __set_connection_args()
```
This is a secret sauce

- `function foo { ... }` or `foo() { ... }` → normal function, runs in the current shell.
- `foo() ( ... )` → runs the body in a subshell (like running in (...)).

azat replied:
It is actually not a secret at all, the difference between `()` and `{}` is known, what confuses me is that you  can have "named" subshells

```
foo
(
)
```
